pkg/clustermesh/common/clustermesh.go

```diff
@@ -270,3 +282,21 @@ func (cm *clusterMesh) ForEachRemoteCluster(fn func(RemoteCluster) error) error
 
 	return nil
 }
+
+const (
+	ClusterMeshModeClusterMeshAPIServer = "clustermesh-apiserver"
+	ClusterMeshModeETCD                 = "etcd"
+	ClusterMeshModeKVStoreMesh          = "kvstoremesh"
+)
+
+// ClusterMeshMode returns the connection type made to the remote cluster.
+func ClusterMeshMode(rcc types.CiliumClusterConfig) string {
+	switch {
+	case rcc.Capabilities.Cached:
+		return ClusterMeshModeClusterMeshAPIServer
```

Comment:
`ClusterMeshModeClusterMeshAPIServer` and `ClusterMeshModeKVStoreMesh` are inverted.